.github/workflows/TestEnv-backend-BuildImageAndPush.yml

```diff
@@ -37,42 +38,32 @@ jobs:
         id: checkout
         uses: actions/checkout@v2
         with:
-          ref: main
+          ref: ${{ github.event.inputs.branch }}
 
-      # Determine the tags
+      # Determine the tags. Create all tags with full names. Only 'tags' are used as this allows specifying
+      # the image name in each tag. This is needed because it is different in each namespace as it contains '-test' suffix in the test namespace.
+      # Note, with multiple tags all have the same digest value.
       - name: Determine image tags
         if: env.IMAGE_TAGS == ''
         run: |
-          echo "IMAGE_TAGS=latest ${GITHUB_SHA::12} account-transfer-test" | tee -a $GITHUB_ENV
+          echo "IMAGE_TAGS=${{ env.TOOLSNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}:latest ${{ env.TOOLSNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}:${GITHUB_SHA::12} ${{ env.TOOLSNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}:account-transfer-test ${{ env.TESTNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}${{ env.TEST_SUFFIX }}:latest ${{ env.TESTNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}${{ env.TEST_SUFFIX }}:${GITHUB_SHA::12} ${{ env.TESTNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}${{ env.TEST_SUFFIX }}:account-transfer-test" | tee -a $GITHUB_ENV
```

Comment:
do we need 'latest'  tags? 'account-transfer-test' will help to see which one is deployed to test at anytime.
	  for easier readabilitity we can do something like (hope it does not over ride env varaible:
	  echo "IMAGE_TAGS=${{ env.TOOLSNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}:${GITHUB_SHA::12} "| tee -a $GITHUB_ENV| tee -a $GITHUB_ENV
	  echo "${{ env.TOOLSNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}:account-transfer-test" | tee -a $GITHUB_ENV 
	  echo "${{ env.TESTNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}${{ env.TEST_SUFFIX }}:${GITHUB_SHA::12}" | tee -a $GITHUB_ENV 
	  echo "${{ env.TESTNS_IMAGE_REGISTRY }}/${{ env.IMAGE_NAME }}${{ env.TEST_SUFFIX }}:account-transfer-test" | tee -a $GITHUB_ENV

Reply:
we can use similar in dev workflows as well